File: C++/Practice/Sem-Finals/Practicals/static.cpp
#include <iostream>
using namespace std;
class base 
{
    public:
    static int stno;
    int no;
    static void staticfunc()
    {
        cout<<"\nThis is a Static Variable and has access to Static Variables = "<<stno;
    }
    void normalfunc()
    {
        cout<<"\nThis is a Normal function which has access to both the normal variable and the static variable "<<no<<" and "<<stno;
    }
};
int base::stno;
int main()
{
    base obj;
    obj.staticfunc();
    obj.normalfunc();
    return 0;
}

File: Java/Practice/UnitTest/static.java
class base 
{
	static int a = 10;
}
class child extends base 
{
	int a = 20;
	public static void main(String args[])
	{
		base obj = new base();
		base obj2 = new base();
		System.out.println(" Value of Static a for first object is = "+obj.a);
		obj.a = obj.a+1;
		System.out.println(" Value of Static a for first object is = "+obj.a);
	}
}
